caikit_tgis_backend/tgis_backend.py

```diff
@@ -195,6 +197,31 @@ def get_connection(
                 self._model_connections.setdefault(model_id, model_conn)
         return model_conn
 
+    def register_model_connection(
```

Comment:
I'm uncertain this `register_model_connection` method is needed at all.

Currently `.get_client()` calls `.get_connection()` with its default arguments which results in the model connection to be created via `TGISConnection.from_config()` and added to `self._model_connections`.

ie. the connection will already be created and added to the _model_connections dict by default if it doesn't exist.